tests/test_type/test_array2.py

```diff
@@ -444,3 +444,18 @@ class Foo(m.Circuit):
     assert not caplog.records, "Should not report an error"
     _check_compile("test_array2_mixed_direction_slice", Foo, False,
                    True)
+
+
+def test_slice_instref(caplog):
+    class Bar(m.Circuit):
+        io = m.IO(I=m.In(m.Array[2, m.Bit]), O=m.Out(m.Array[2, m.Bit]))
+
+    class Foo(m.Circuit):
+        T = m.Array[4, m.Bit]
+        io = m.IO(I=m.In(T), O=m.Out(T))
+        bar = Bar()
+        bar.I @= io.I[2:]
+        io.O[2:] @= bar.O
+        io.O[:2] @= bar.O
+        assert list(io.O.trace().connection_iter()) == [
+            (bar.O, None), (bar.O, None)]
```

Comment:
why is the second element None? shouldn't it be io.O[slice(0, 2)]? (or slice(2, 4))?

Reply:
Trace returns the driver, the driver itself is not connected (not driven by anything)

Reply:
(Connection iter is a recursive trace over the children)

Reply:
i thought connection x.iter returns (driver of x_i, x_i) where x_i can be slice or element or whole array

Reply:
Yes this calls connection iter on the return value of a trace call, not the original value

Reply:
ooh missed that, ok makes sense. so does io.O.connection_iter returns [(bar.O, io.O[slice(0, 2)), (bar.O, io.O[slice[2, 4])]?

I'm just trying to understand what this fixes. What would have been different w.r.t. this test before this fix?

Reply:
`io.O.trace()` returns: `m.array([io.I[slice(2, 4)], io.I[slice(0, 2)])`
calling connection iter on that anon value, io.I is not driven

Reply:
I clarified the test by using `driven` (which covers the enumerate children code path) instead of `trace`